sale_stock_restocking_fee_invoicing/models/stock_move.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from collections import defaultdict
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    charge_restocking_fee = fields.Boolean(
+        help="If checked your customer will be charged for accepting "
+             "returned goods.",
+        default=False,
+        readonly=True,
+    )
+    is_customer_return = fields.Boolean(
+        related="picking_id.is_customer_return", readonly=True
+    )
+
+    @api.multi
+    def action_done(self):
+        result = super(StockMove, self).action_done()
+        move_ids_to_charge_by_so = defaultdict(list)
+        for move in self:
+            if not move.charge_restocking_fee:
+                continue
+            if not move.origin_returned_move_id:
+                continue
```

Comment:
Isn't shorter this?:
```suggestion
        for move in self.filtered(lambda x: x.charge_restocking_fee and x.origin_returned_move_id):
```

Reply:
IMO, the readability is better without filtered. In this case, Moreover in this case, We don't need to reduce the recordset on which the process is done.

Reply:
Well, at least join both conditions:
```suggestion
        for move in self:
            if not move.charge_restocking_fee or not move.origin_returned_move_id:
                continue
```

Reply:
Matter of style I guess. Personally I'd just go with

```python
if move.charge.. and move.origin..:
 do_stuff()
```
I get sometimes it works doing a continue if there is a massive block of code underneath with more branches, but it is 2 lines.

Reply:
Yeah, indeed this is not blocking. I don't like that style though, as it implies an extra level of indentation for your main code, although it avoids uncovered lines in tests. We never have the ideal one!